src/execution/vm/bytecode_generator.cpp

```diff
@@ -1711,15 +1733,36 @@ void BytecodeGenerator::VisitBuiltinTrigCall(ast::CallExpr *call, ast::Builtin b
 }
 
 void BytecodeGenerator::VisitBuiltinArithmeticCall(ast::CallExpr *call, ast::Builtin builtin) {
-  LocalVar dest, src;
-  auto dest_type = call->GetType();
-  dest = GetExecutionResult()->GetOrCreateDestination(dest_type);
-  src = VisitExpressionForRValue(call->Arguments()[0]);
-  Bytecode abs_bytecode = call->Arguments()[0]->GetType()->IsIntegerType() ||
-                                  call->Arguments()[0]->GetType()->IsSpecificBuiltin(ast::BuiltinType::Integer)
-                              ? Bytecode::AbsInteger
-                              : Bytecode::AbsReal;
-  GetEmitter()->Emit(abs_bytecode, dest, src);
+  LocalVar dest = GetExecutionResult()->GetOrCreateDestination(call->GetType());
+
+  switch (builtin) {
+    case ast::Builtin::Abs: {
+      LocalVar src = VisitExpressionForRValue(call->Arguments()[0]);
+      Bytecode abs_bytecode = call->Arguments()[0]->GetType()->IsIntegerType() ||
```

Comment:
Change to:
```
Bytecode bytecode = call->Arguments[0]->GetType()->IsSpecificBuiltin(ast::BuiltinType::Integer) ? Bytecode::AbsInteger : Bytecode::AbsReal
```

`IsInteger()` checks if the type is a primitive integer (i.e., not a SQL value).

Reply:
Should I modify the semantic checking as well? Because that also checks for both primitive and SQL value.